maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -324,7 +325,8 @@ private void doExecute2(MavenSession session, MojoExecution mojoExecution) throw
             } catch (MojoFailureException
                     | PluginManagerException
                     | PluginConfigurationException
-                    | MojoExecutionException e) {
+                    | MojoExecutionException
+                    | MojoException e) {
```

Comment:
Should that one be a `MavenException` too given we've broaden the other locations ?

Reply:
Yes, but the intent is that the code either calls `MojoSucceeded` or `MojoFailed`.  Since any `MavenException` can be thrown at this point, they need to be caught imho.